### PR TITLE
feat: General LedgerClient features

### PR DESCRIPTION
**Pull request by MissingNO57**
## Proposed changes

Added features to LedgerClient that are available on CosmJS

Features:
- Query the latest block
- Query block by height
- Query chain id
- Query current height

Example:
```
    ledger = LedgerClient(NetworkConfig.fetchai_stable_testnet())

    block = ledger.query_block(6662820)
    tx_hash = block.tx_hashes[0]

    tx = ledger.query_tx(tx_hash)

    print(f"tx: {tx}")
```

## Issues

Links to any issues resolved.

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to stop working as expected)
- [ ] Something else (e.g. tests, scripts, example, deployment, infrastructure, ...)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](https://github.com/fetchai/cosmpy/blob/main/CONTRIBUTING.md) document.
- [x] I have based my branch, and I am making a pull request against, the `main` branch.
- [x] Checks and tests pass locally.

### If applicable

- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have checked that code coverage does not decrease.
- [x] I have added/updated the documentations.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did, what alternatives you considered, etc.

### CODE REVIEW

File: cosmpy/aerial/client/__init__.py [hunk 1]
```diff
@@ -716,3 +741,57 @@ def broadcast_tx(self, tx: Transaction) -> SubmittedTx:
...
+        resp = self.tendermint.GetBlockByHeight(req)
+        return self._parse_block(resp)
+
+    def _parse_timestamp(self, timestamp: Timestamp) -> datetime:
+        """Parse the timestamp.
+
+        :param timestamp: timestamp
+        :return: parsed timestamp
+        """
+        return datetime.fromtimestamp(timestamp.seconds)
```
Comment (MissingNO57) [inline cosmpy/aerial/client/__init__.py:770]: timestamp has a microsecond part, should we consider it?

File: cosmpy/aerial/client/__init__.py [hunk 2]
```diff
@@ -716,3 +741,57 @@ def broadcast_tx(self, tx: Transaction) -> SubmittedTx:
...
+    def _parse_block(self, block: GetBlockByHeightResponse) -> Block:
+        """Parse the block.
+
+        :param block: block as GetBlockByHeightResponse
+        :return: parsed block as Block
+        """
+        return Block(
+            height=int(block.block.header.height),
+            time=self._parse_timestamp(block.block.header.time),
+            tx_hashes=[sha256(tx).hex().upper() for tx in block.block.data.txs],
```
Comment (MissingNO57) [inline cosmpy/aerial/client/__init__.py:781]: We can get Tx here, not the tx response 

- Tx is in bytes format which can be deserialised into Tx protobuf object - but it has no hash value anyway.
The only solution for txhash I found is to just fo sha256 - You can query node using this hash to get TxResponse.

**Review by jrriehl — APPROVED** [review]
Nice!

File: cosmpy/aerial/client/__init__.py (hunk 2 above)
Comment (pbukva) [reply cosmpy/aerial/client/__init__.py]: The best solution would be to create dedicated method for Tx hash derivation in comspy, whic would be able to take as input parameter of various types (e.g. `bytes` (=Tx serialised in protobuf encoding) and  `Tx` & `TxRaw` python types generated from CosmosSDK protobuf schema (`github.com/cosmos/cosmos-sdk/types/tx`).

But for now, th in-place solution provided here is sufficient.

File: cosmpy/aerial/client/__init__.py [hunk 3]
```diff
@@ -172,6 +184,41 @@ def total_unbonding(self) -> int:
...
+    @staticmethod
+    def _parse_timestamp(timestamp: Timestamp):
+        """Parse the timestamp.
+
+        :param timestamp: timestamp
+        :return: parsed timestamp
+        """
+        return datetime.fromtimestamp(timestamp.seconds, tz=timezone.utc) + timedelta(
+            microseconds=timestamp.nanos // 1000
+        )
```
Comment (pbukva) [inline cosmpy/aerial/client/__init__.py:219]: It kind of feels that this method would deserve to me relocated to a place with more generic purpose than `Block` class - normally it ould be somethiong like `utils`, `common` or similar, but it looks like that cosmpy does not have such explicitly named  components and rather uses `__init__.py` files to do such generic purpose things.

Comment (MissingNO57) [reply cosmpy/aerial/client/__init__.py:219]: I had the same feeling, but we should move it together with other stuff in some refactoring PR.

**Pull request merged**